WordPress/Classes/Utility/Universal Links/RouteMatcher.swift

```diff
@@ -61,9 +61,22 @@ class RouteMatcher {
             values[MatchedRouteURLComponentKey.fragment.rawValue] = fragment
         }
 
+        if let source = sourceQueryItemValue(for: url) {
+            values[MatchedRouteURLComponentKey.source.rawValue] = source
+        }
+
         return values
     }
 
+    private func sourceQueryItemValue(for url: URL) -> String? {
```

Comment:
Just a matter of preference (can be ignored): you can just do a single

`URLComponents(url: url, resolvingAgainstBaseURL: false)?.queryItems?.first(where: { $0.name == "source" })?.value?.removingPercentEncoding`

Reply:
Sure, thank you – I just preferred splitting things out a little so there's not quite so much on one line here.